Vineyard/Vineyard/CreateGroupView.swift

```diff
@@ -70,12 +98,41 @@ struct CreateGroupView: View {
                         Image(systemName: "xmark.circle.fill")
                     }
                 }
+            }.navigationDestination(isPresented: $isValid) {
+                GoalsListView(groupName: $groupName, resolution: $resolution, deadline: $deadline)
             }
         }
 
     }
 }
 
+
+func validateForm(groupName: String, resolution: String, deadline: Date) throws -> Bool {
```

Comment:
We should never have high level functions. This can also go into the view model, including the structs you have below.

Reply:
moved to viewModel